File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        res=[]
        for i in range(len(nums)):
            sum=0
            for j in range(0,i+1):
                sum=sum+nums[j]
            res.append(sum)
        return res
            

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.java
class Solution {
    public static void func(int[] nums,int index){
        if(index==nums.length){
            return;
        }
        nums[index]=nums[index]+nums[index-1];
        func(nums,index+1);
    }
    public int[] runningSum(int[] nums) {
        func(nums,1);
        return nums;
    }
}